apps/api/views/metrics.py

```diff
@@ -0,0 +1,761 @@
+                        outsider_contrib = {
+                            'cha_username': 'others',
+                            'commit_count': outsider_commit_count,
+                        }
+                        contributor_metrics.append(outsider_contrib)
+                    data = {
+                        'repo_url': latest_submission.github_url,
+                        'github_contributions': contributor_metrics,
+                        'chagrade_submissions': team_submissions,
+                    }
+                    return data
+                else:
+                    return 'Team leader has no submissions.'
+            else:
+                return 'Team leader not connected to Github.'
+        else:
+            return 'No team leader.'
+
+
+@api_view(['GET'])
+@permission_classes(())
+def chagrade_overall_metrics(request, version):
+    data = {
+        "users": ChaUser.objects.count(),
+        "students": StudentMembership.objects.count(),
+        "instructors": Instructor.objects.count(),
+        "submissions": Submission.objects.count(),
+        "klasses": Klass.objects.count(),
+    }
+    return Response(data, status=status.HTTP_200_OK)
+
+
+#################################################
+################# Admin Metrics #################
+#################################################
+
+
+class SubmissionMetricsView(TimeSeriesObjectCreationQueryMixin, ScoreDistributionMixin, APIView):
+    permission_classes = (InstructorOrSuperuserPermission,)
+    time_series_model = Submission
+    time_series_creation_date_field_name = 'created'
+
+    def get(self, request, **kwargs):
+        score_distribution = self.score_distribution_query()
+
+        submissions = self.time_series_query()
+        output_data = {
+            'submissions_made': submissions,
+            'submission_scores': score_distribution,
+        }
+        return Response(output_data)
+
+
+class StudentMetricsView(TimeSeriesObjectCreationQueryMixin, APIView):
+    permission_classes = (InstructorOrSuperuserPermission,)
+    time_series_model = StudentMembership
+    time_series_creation_date_field_name = 'date_enrolled'
+
+    def get(self, request, **kwargs):
+        users = self.time_series_query()
+        return Response(users)
+
+
+class InstructorMetricsView(TimeSeriesObjectCreationQueryMixin, APIView):
+    permission_classes = (InstructorOrSuperuserPermission,)
+    time_series_model = Instructor
+    time_series_creation_date_field_name = 'date_promoted'
+
+    def get(self, request, **kwargs):
+        instructors = self.time_series_query()
+        return Response(instructors)
+
+
+class KlassMetricsView(TimeSeriesObjectCreationQueryMixin, APIView):
+    permission_classes = (InstructorOrSuperuserPermission,)
+    time_series_creation_date_field_name = 'created'
+    time_series_model = Klass
+
+    def get(self, request, **kwargs):
+        klasses = self.time_series_query()
+
+        class Round(Func):
```

Comment:
What's going on with this?

Reply:
This is used to round to two decimal places in the database, so I don't have to create a loop to round all of the averages after querying.